core/src/main/java/org/elasticsearch/common/util/CancellableThreads.java

```diff
@@ -92,6 +94,7 @@ public void execute(Interruptable interruptable) {
      *
      * @param interruptable code to run
      */
+    @SuppressForbidden(reason = "Use Thread.interrupted() to implement this class' interruption policy (forbid external interrupts).")
```

Comment:
I'm not particularly happy that you have to suppress *all* forbidden checks on this method just to suppress the one that you want.